shipyard_blueprints/googledrive/shipyard_googledrive/drive_utils.py

```diff
@@ -269,15 +287,146 @@ def get_file_matches(
+                            driveId=drive_id,
+                        )
+                        .execute()
+                    )
+                else:
+                    results = (
+                        service.files()
+                        .list(
+                            supportsAllDrives=True,
+                            includeItemsFromAllDrives=True,
+                            corpora="drive",
+                            driveId=drive_id,
+                        )
+                        .execute()
+                    )
+            else:
+                if query:
+                    results = service.files().list(q=query).execute()
+                else:
+                    results = service.files().list().execute()
+
+            files = results.get("files", [])
+        else:
+            files = []
+            all_folder_ids = get_all_folder_ids(service, drive_id=drive_id)
+            for f_id in all_folder_ids:
+                # query += f" and '{f_id}' in parents"
+                query = f"'{f_id}' in parents"
+                if drive_id:
+                    results = (
+                        service.files()
+                        .list(
+                            q=query,
+                            supportsAllDrives=True,
+                            includeItemsFromAllDrives=True,
+                            corpora="drive",
+                            driveId=drive_id,
+                        )
+                        .execute()
+                    )
+                else:
+                    results = service.files().list(q=query).execute()
+
+                files.extend(results.get("files", []))
+
+            # grab the files in the root
+            if drive_id:
+                root_results = (
+                    service.files()
+                    .list(
+                        q="trashed=false and mimeType!='application/vnd.google-apps.folder'",
+                        supportsAllDrives=True,
+                        includeItemsFromAllDrives=True,
+                        corpora="drive",
+                        driveId=drive_id,
+                    )
+                    .execute()
+                )
+            else:
+                root_results = (
+                    service.files()
+                    .list(
+                        q="trashed=false and mimeType!='application/vnd.google-apps.folder'"
+                    )
+                    .execute()
+                )
+
+            files.extend(root_results.get("files", []))
+
+        matches = []
+        id_set = set()
+        for f in files:
+            if re.search(pattern, f["name"]) and f["id"] not in id_set:
+                matches.append(f)
+                id_set.add(f["id"])
     except Exception as e:
         raise ExitCodeException(f"Error in finding matching files: {str(e)}", 210)
 
     else:
-        return results.get("files", [])
+        return matches
+
+    # if folder_id:
```

Comment:
Do we want to remove this commented out section or did you leave it in on purpose?